### ISSUE TITLE
More explicit bookmarks name

### ISSUE THREAD

**Issue by sucrecacao** [jarun/nnn#553]
**Is your feature request related to a problem? Please describe.**
Bookmark name are 1 character long, which make it hard to remember which char correspond to which bookmark.

**Describe the solution you'd like**
I see to possible solution:
1) We can use longer name in the `NNN_BMS` variable.
```bash
export NNN_BMS="media:/path/to/media"
```
When pressing the bookmark key the full name `media`  is printed. The first character of the bookmark name is implicitly the key that need to be pressed ( `m` in this case ) .
We can change the name like `d-media` if we want to use `d` instead of `m` as the key.

2) like the above, exept that the key must be specified between quotes:
```bash
export NNN_BMS="'m'edia:/path/to/media"
```
So we can use keys that are not the first letter of the bookmark name:
```bash
export NNN_BMS="'m'edia:/path/to/media;mo'v'ie:/path/movies"
```
If the bookmark is of length 1, we fallback on the current nnn behaviour. 

Both option are compatible with the current `nnn` behaviour.
I'm in favour of option 2. I think it's more explicit.

We could also have both option 1 and 2 at the same time.
Option 2 could also fallback on option 1 if the bookmark name contain more than one letter but doesn't contain quotes. But maybe it's better to have only one way of doing stuff.

**Comment by 0xACE** [comment]
I can't predict if this will be adopted.

But fyi, you can make a plugins that lists these details for you... infact iirc we implemented for your plugins to be able to control where `nnn` browses to, so you could replace this entire issue with a plugin that handles exactly this....

**Comment by jarun** [comment]
1. The current thing is simple and straight
2. We cannot re-work this at this stage of the project. This would also lead to re-working plugins. I don't want them handled ddifferently.

If your main issue is immediate remembering (because you'll remember in the long run by habit anyway), always visit the help and config page where we list the bookmarks.

Also, @toddyamakawa shared an idea of a bookmark plugin whch would list it the way you want. Can we have it in @toddyamakawa?

**Comment by toddyamakawa** [comment]
> Also, @toddyamakawa shared an idea of a bookmark plugin whch would list it the way you want. Can we have it in @toddyamakawa?

@sucrecacao, it's easily doable with a plugin. I don't use nnn's built-in bookmarks:
https://github.com/toddyamakawa/bin/blob/master/nnn-bookmarks

@jarun @leovilok I was just testing my plugin and it looks like that `nnn` isn't properly handling my `NNN_PIPE`. Any idea what happened?

**Comment by jarun** [comment]
> it looks like that nnn isn't properly handling my NNN_PIPE

We have a spec in place now. ;) See https://github.com/jarun/nnn/tree/master/plugins#send-data-to-nnn

    printf "%s" "0$(readlink -f "$cddir")" > "$NNN_PIPE"
has to change:

**Comment by jarun** [comment]
Also, please add it as a plugin and document what it does.

**Comment by toddyamakawa** [comment]
> We have a spec in place now. ;) See https://github.com/jarun/nnn/tree/master/plugins#send-data-to-nnn

We can send things to different contexts?? This is awesome! :)

> Also, please add it as a plugin and document what it does.

Sounds good. I'll work on it after the live preview

**Comment by jarun** [comment]
:+1:

**Comment by jarun** [comment]
Implemented at PR #558.

**Comment by jarun** [comment]
@sucrecacao symlinked bookmarks is now implemented in the core program. Will be available on the next release.
